ts/src/flexible-event/privacy.ts

```diff
@@ -159,10 +171,35 @@ export function maxInformationGain(numStates: number, epsilon: number): number {
   )
 }
 
+export function informationGainWithAttributionScopes(
+  numStates: number,
+  attributionScopeLimit: number,
+  maxEventStates: number
+): number {
+  return Math.log2(numStates + maxEventStates * (attributionScopeLimit - 1))
```

Comment:
What happens here when `numStates === 0 && attributionScopeLimit === 1`? Please add a test case to that effect.

Reply:
`numStates` should be always positive. but yes, you are right, we shouldn't make this assumption here.

Reply:
Do we need to handle this case in the spec as well?